### DISCUSSION TITLE
redhat.java-1.14.0-win32-x64 is not recognized as installed ; jre folder missing

### DISCUSSION THREAD

**Discussion by jcrben** [redhat-developer/vscode-java#2893] [General]
I initially was going to post an issue about this, but I managed to figure it out and I figure I might as well post something in case it helps someone else.

### issue resolution tl;dr
My redhat.java auto-updated from 1.13.0 to 1.14.0 and when it did that, the jre folder was missing inside of it. Possibly a network failure or maybe it used an endpoint which was firewalled. I typically install via the cli in a script I've written. I reinstalled 1.13.0 to get the jre folder back then tried 1.14.0 and managed to get a jre folder when installing from the cli. 

However when I reinstalled 1.14.0 VSCode did not recognize that install so I reinstalled 1.13.0 at the cli and it did recognize that install. So currently my extension seems to be fixed and I'll be disabling autoupdate for now.

### Troubleshooting steps long version
I reinstalled 1.13.0 via `code --install-extension redhat.java@1.13.0` and found that it did have the jre. Ultimately I ran it again for 1.14.0 (`code --force --install-extension redhat.java@1.14.0`) to take 1.14.0.

In between I ran that command improperly once (`code --install-extension redhat.java@1.13`) - omitting the last version - and also renamed the folder for these which might have caused the issue.

Steps I tried to fix it:
* click the "install and reload" prompt that I get because my various other java projects want it installed. Get the same prompt after reload
* deleted all the caches inside of C:\Users\bncrsy\AppData\Roaming\Code
* note: did not delete any local storage etc, only caches
* 
This happened on a corporate computer where I don't have total freedom to mess around too much or provide too much information about my system or code.

**Comment by rgrunber** [comment]
Is this still an issue even now ?

When I look at what's on the VS Code Marketplace for 1.14.0 win32-x86_64, I can see `extension/jre/17.0.5-win32-x86_64/bin/java.exe` under that version's vsix. Maybe the update somehow took place before the extension payload was in place ? I would think this doesn't happen normally though.

There are a few interesting things that happened in our 1.14.0 release though :

- https://github.com/redhat-developer/vscode-java/pull/2854 ( I don't think this would have caused the issue)
- https://github.com/redhat-developer/vscode-java/pull/2880/files , where we went from publishing 10 vsix (5 with jre, 5 without), to publishing 6 vsix (5 with jre, 1 without).

The version without the JRE should only be installed on a system that is **not** one of the following : `win32-x64`, `linux-x64`, `linux-arm64`, `darwin-x64`, `darwin-arm64`. Only thing I can think of, assuming your system is win32-x64, is maybe VS Code somehow failed to detect that and installed the universal vsix (without the JRE).
